### PR TITLE
ColorPicker: Cleanly implement large size

### PR DESCRIPTION
**Pull request by mirka**
## What?

Replaces the hacky implementation of the 40px size.

## Why?

We now have large size variants on the underlying components that can be used cleanly via props.

## Testing Instructions

1. `npm run storybook:dev`
2. See the story for ColorPicker. There should be no visual changes in all modes (Hex, RGB, HSL).

### CODE REVIEW

**Review by ciampo — APPROVED** [review]
It's great to see the results of previous work, code definitely looks cleaner!

The only side-effect of this change is the horizontal padding in the input fields — previously it was `8px`, while with the changes from this PR, it becomes `16px` (see [source code](https://github.com/WordPress/gutenberg/blob/be92ce8d06f1e282e9cfb401e43790d60ef8f137/packages/components/src/input-control/styles/input-control-styles.tsx#L148-L180))

| trunk | this PR |
|---|---|
| Screenshot 2022-07-01 at 12 40 21 |  Screenshot 2022-07-01 at 12 40 08 |

I'm going to approve this PR, as I think still think this is fine for the sake of applying a more uniform design across our components (cc @jameskoster in case you have any feedback).

**Comment by jameskoster** [comment]
I say this with no understanding of the impact on the underlying component(s), or how best to proceed, but that spacing does feel a bit large to me :) An 8px gap between prefix and input value seems more natural. 

The 16px gap on the right is good. It would be nice to match the prefix spacing with that, currently it is 14px for some reason 🤔

**Comment by mirka** [comment]
Great catch @ciampo, and thanks for the check @jameskoster. I proposed #42166 for better default padding in the prefix/suffix case. If that looks good, let's wait for that to land before merging this PR.

> The 16px gap on the right is good. It would be nice to match the prefix spacing with that, currently it is 14px for some reason

Will do 👍

**Comment by mirka** [comment]
Spacings looking good after #42166 ✨ 

R/G/B prefixes in the ColorPicker with correct spacing

**Pull request merged**
